optaplanner-core/src/main/java/org/optaplanner/core/api/solver/manager/SolverManager.java

```diff
@@ -0,0 +1,271 @@
+     *                                   (which might or might not be optimal, feasible or even initialized). It can take seconds, minutes,
+     *                                   even hours or days before this method is invoked, depending on the {@link Termination} configuration and
+     *                                   scheduling of the solver job
+     * @param onException                sometimes null, a callback invoked if the solver job throws an exception
+     */
+    void solve(
+            Object problemId,
+            Solution_ planningProblem,
+            Consumer<Solution_> onBestSolutionChangedEvent,
+            Consumer<Solution_> onSolvingEnded,
+            Consumer<Throwable> onException);
+
+    /**
+     * Notifies the solver that it should stop at its earliest convenience and clean up all the resources used by the
+     * corresponding solver task. Note that after calling this method there will be no reference for problemId in the
+     * {@link SolverManager}.
+     *
+     * @param problemId never null, a unique id for each planning problem
+     * @return true if successful, false if no problem with problemId has been submitted
+     */
+    boolean terminateSolver(Object problemId);
+
+    /**
+     * Notifies the solver that it should stop at its earliest convenience.
+     * To resume solving call {@link #resumeSolver(Object problemId)}
+     *
+     * @param problemId never null, a unique id for each planning problem
+     * @return true if successful, false if no problem with problemId has been submitted
+     */
+    boolean pauseSolver(Object problemId);
+
+    /**
+     * Resumes solving a previously stopped solver task.
+     *
+     * @param problemId never null, a unique id for each planning problem
+     * @return true if successful, false if no problem with problemId has been submitted or solver is already solving
+     */
+    boolean resumeSolver(Object problemId);
+
+    /**
+     * @param problemId never null, a unique id for each planning problem
+     * @return true if a planning problem with id {@param problemId} has already been submitted via {@link #solve}
+     */
+    boolean isProblemSubmitted(Object problemId);
+
+    /**
+     * The best solution is the {@link PlanningSolution best solution} found during solving:
+     * it might or might not be optimal, feasible or even initialized.
+     * <p>
+     * This method is useful in rare asynchronous situations (although {@code onBestSolutionChangedEvent} callback
+     * of {@link #solve} is often more appropriate).
+     *
+     * @param problemId never null, a unique id for each planning problem
+     * @return null if the problemId isn't associated with a previously submitted problem, or
+     * {@link PlanningSolution best solution} which might be uninitialized with a null {@link Score} if the solver job
+     * has not started yet
+     */
+    Solution_ getBestSolution(Object problemId);
```

Comment:
This method causes a fundamental memory leak, by design.
It was part of the requirements, but we need to rethink this.
Let's discuss on Thursday.